src/components/MapComponent.js

```diff
@@ -10,124 +10,123 @@ const MapComponent = () => {
 
-    useEffect(() => {
-        let isMounted = true; // Track if the component is mounted
-
-        const fetchLocationAndInitializeMap = async () => {
-            // Fetch user's location based on IP, default to Bellevue if not found
-            let userLat = 47.610378;
-            let userLng = -122.200676;
-
-            try {
-                const response = await fetch(`https://ipinfo.io/json?token=${process.env.REACT_APP_IP_INFO_TOKEN}`);
-                if (response.ok) {
-                    const data = await response.json();
-                    const loc = data.loc.split(',');
-                    userLat = parseFloat(loc[0]);
-                    userLng = parseFloat(loc[1]);
-                } else {
-                    console.error('Failed to fetch location:', response.statusText);
-                }
-            } catch (error) {
-                console.error('Error fetching location:', error);
+    // Function to generate random points around a place
+    const generateRandomPoints = (center, numPoints, maxDistance, maxIntensity) => {
+        const points = [];
+        for (let i = 0; i < numPoints; i++) {
+            const angle = Math.random() * Math.PI * 2;
+            const distance = Math.random() * maxDistance;
+            const lat = center[0] + (distance / 111) * Math.cos(angle); // 111 km per degree of latitude
+            const lng = center[1] + (distance / (111 * Math.cos(center[0] * (Math.PI / 180)))) * Math.sin(angle); // Adjust for longitude
+            const intensity = Math.random() * maxIntensity;
+            points.push([lat, lng, intensity]);
+        }
+        return points;
+    };
+
+    function addHeatLayer(map) {
+        // Generate N random points around Bellevue with intensity between 0 and 1
+        const heatData = generateRandomPoints([47.610378, -122.200676], 1000, 5, 1);
+
+        // Add the heat layer to the map
+        L.heatLayer(heatData, {
+            radius: 25,      // Radius of each “point” of the heatmap
+            blur: 15,        // Amount of blur
+            maxZoom: 18      // Zoom level at which the points reach maximum intensity
+        }).addTo(map);
+    }
+
+    async function fetchUserLocation() {
+        // Fetch user's location based on IP, default to Bellevue if not found
+        let userLat = 47.610378;
+        let userLng = -122.200676;
+
+        try {
+            const response = await fetch(`https://ipinfo.io/json?token=${process.env.REACT_APP_IP_INFO_TOKEN}`);
+            if (response.ok) {
+                const data = await response.json();
+                const loc = data.loc.split(',');
+                userLat = parseFloat(loc[0]);
+                userLng = parseFloat(loc[1]);
+            } else {
+                console.error('Failed to fetch location:', response.statusText);
             }
-
-            // Proceed to initialize the map if the component is still mounted
-            if (isMounted) {
```

Comment:
Yes the reinitialization was because you didnt pass dependency array in useEffect. I passed an empty array which will only cause it to run once. Hence solving our problem and keeping the code cleaner.

Reply:
I see, thanks for the explanation! Guess multi-initialization also related to the React.StrictMode you mentioned above, yeah we can remove this mount state check as we removed the StrictMode for now.